### ISSUE TITLE
Test `texture-compression-bc-sliced-3d`

### ISSUE THREAD

**Issue by mehmetoguzderin** [gpuweb/cts#3761]
https://github.com/gpuweb/gpuweb/issues/4705

`webgpu:api,validation,texture,*`, and also operation tests where applicable.

**Comment by kainino0x** [comment]
I think we also need to make sure we have basic sampling operation tests, as well as both validation and operation tests for copyTextureToTexture/TextureToBuffer/BufferToTexture and writeTexture.

**Comment by mehmetoguzderin** [comment]
@kainino0x Thank you very much for the guidance, I missed those aspects. I will follow-up with a PR for those tests, too! Appreciated!

**Comment by kainino0x** [comment]
@greggman since you just added 2d compressed texture sampling tests in #3760, would it make sense for you to do the sampling tests for 3d? It will probably fit into what you're already working on.

**Comment by greggman** [comment]
Yes, sure. I can either do it now and they'll fail or we can add them later but the code will just be almost copy and pasted from the 2d stuff so unless I remove them they'll already kind of maybe do the right thing but fail until implemented.

**Comment by mehmetoguzderin** [comment]
Change needs to cover not only validation but also operation, reopening. Thank you.

**Comment by mehmetoguzderin** [comment]
It seems like https://github.com/gpuweb/cts/commit/cb42984fa7f024120ce4731120007856e33f9d6a actually fixed the additional limitation in util/texture/base, and there doesn't seem to be any additional particular filters to not test copies or writes of compressed 3D formats. I'll create a draft PR to dry run. Thank you!

**Comment by greggman** [comment]
so fyi, the texture builtin tests should be able to test this stuff. Most of the exist tests just filter out compressed 3d (as slices) texture formats. If/when want to turn them on we just need to change the filters.

**Comment by mehmetoguzderin** [comment]
@greggman thank you very much for the information.

**Comment by beaufortfrancois** [comment]
> so fyi, the texture builtin tests should be able to test this stuff. Most of the exist tests just filter out compressed 3d (as slices) texture formats. If/when want to turn them on we just need to change the filters.

@greggman Would you be able to help on that front now that `texture-compression-bc-sliced-3d` is going to be available in Chromium? See https://chromium-review.googlesource.com/c/chromium/src/+/6459626

**Comment by greggman** [comment]
Here's a PR

https://github.com/gpuweb/cts/pull/4359

**Comment by beaufortfrancois** [comment]
You are quick! Thank you @greggman!

**Comment by beaufortfrancois** [comment]
@greggman Once is https://github.com/gpuweb/cts/pull/4359 is reviewed and merged, can you advise on next steps?

**Comment by greggman** [comment]
theoretically there are no next steps. it should work 😇🙏💥😂

if it doesn't I can fix it. I'm out of office until April 28th

**Comment by beaufortfrancois** [comment]
@greggman Does this mean this covers operation tests for the texture-compression-bc-sliced-3d feature?

**Comment by beaufortfrancois** [comment]
@greggman Following https://github.com/gpuweb/cts/pull/4359, the [webgpu:shader,execution,expression,call,builtin,textureSampleBias:sampled_2d_coords:format="bc1-rgba-unorm-srgb";filt="nearest";modeU="r";modeV="m";offset=true](https://gpuweb.github.io/cts/standalone/?debug=1&q=webgpu:shader,execution,expression,call,builtin,textureSampleBias:sampled_2d_coords:format=%22bc1-rgba-unorm-srgb%22;filt=%22nearest%22;modeU=%22r%22;modeV=%22m%22;offset=true) test now fails including the warning message:
```
### WARNING: sample points are derived from un-compressed textures and may not match the
actual GPU results of sampling a compressed texture. The test itself failed at this point
(see expected: and got: above). We're only trying to determine what the GPU sampled, but
we can not do that easily with compressed textures. ###
```
Shall we exclude compressed formats for those tests?

**Comment by beaufortfrancois** [comment]
> I think so. checking, it looks like the various copy tests should test these already too

I've noticed copyTextureToTexture tests needed to be updated as well. I've opened https://github.com/gpuweb/cts/pull/4371 to test this.

**Comment by mehmetoguzderin** [comment]
@beaufortfrancois thank you for the PRs. it seems that upstream tests BC with the merges now, do you think this issue is OK to close? Appreciated.

**Comment by beaufortfrancois** [comment]
I'd like to make sure CTS is rolled propetly in Dawn before. It's not the case yet. See https://dawn-review.googlesource.com/c/dawn/+/239295

**Comment by beaufortfrancois** [comment]
I believe we can mark this issue as fixed now as we have both validation and operation tests.
